lighthouse-core/report/v2/report-renderer.js

```diff
@@ -64,15 +71,31 @@ window.ReportRenderer = class ReportRenderer {
   }
 
   /**
-   * @param {!ReportJSON} report
-   * @return {!Element}
+   * Sets the text content of an element, safely. If the element does not exist,
+   * results in a noop.
+   * @param {Element} element
+   * @param {!string} text
+   * @return {Element}
    */
-  renderReport(report) {
-    try {
-      return this._renderReport(report);
-    } catch (e) {
-      return this._renderException(e);
+  static setText(element, text) {
+    if (element) {
```

Comment:
> @paulirish asked for them so that the report fails gracefully if a node cant be found.

I think @paulirish was asking before we had completely settled on the approach used here, so it may be less of an issue. Queries failing fast (as soon as we try to set `textContent` or add to the `classList`) is a good thing, I think. I also like the idea of moving some of the query verification to tests, but if setting `element.textContent` throws a TypeError, just testing e.g. `_renderAuditScore` will verify the relevant elements exist if it doesn't throw :)